Added serviceaccount configuration (#214)
okteto/docs: src/content/self-hosted/administration/configuration.mdx

      key: "ca.crt"
```

### serviceaccount

Annotation and label customizations for service accounts.
The configured annotations and labels will be applied to each service account created by Okteto.
These annotations and labels are additional to the ones already applied by Okteto.

- `annotations`: Annotations applied to each service account generated by Okteto for user accounts.
- `labels`: Labels applied to each service account generated by Okteto for user accounts.

```yaml

> serviceaccount:

Author: yes, fixed this, the serviceaccount config will be under `user`

      key: "ca.crt"
```

> ### serviceaccount

Reviewer: we use camel case in other settings. we should be consistent here.

> ### user serviceaccount

Reviewer: Is this the format we use for subkeys?

Author: i thought adding user to serviceaccount will determine more the key for the config, but i see a mix of titles for different keys... i will stick to `serviceaccount` because is the config of serviceaccount, and then keep the example with the key of user, WDYT? i'm not sure what would be the best here

> Annotation and label added to every user's service account.

Reviewer: This is redundant with the next sentence.
```suggestion

```

> Annotation and label customizations for users service accounts.

Reviewer: ```suggestion
Annotation and label added to every user's service account.
```

Reviewer: I think this is redundant given that we explain the same on the next line.

      key: "ca.crt"
```

### serviceaccount

> Annotation and label customizations for serviceaccounts.

Reviewer 2: ```suggestion
Annotation and label customizations for `serviceaccounts`.
```

> The configured annotations and labels will be applied to each serviceaccount created by Okteto.

Reviewer 2: ```suggestion
The configured annotations and labels will be applied to each `serviceaccount` created by Okteto.
```

These annotations and labels are additional to the ones already applied by Okteto.

> - `annotations`: Annotations applied to each serviceaccount generated by Okteto.

Reviewer 2: ```suggestion
- `annotations`: Annotations applied to each `serviceaccount` generated by Okteto.
```

> - `labels`: Labels applied to each serviceaccount generated by Okteto.

Reviewer 2: ```suggestion
- `labels`: Labels applied to each `serviceaccount` generated by Okteto.
```

```yaml
serviceaccount:
  annotations:
    custom.annotation/one: one
    custom.annotation/two: two
  labels:
    custom.label/one: one
    custom.label/two: two
```

> Use these custom annotations to add IAM roles to service accounts and enable those service account to access the cluster resources.

Reviewer 2: ```suggestion
Use these custom annotations to add IAM roles to service accounts and enable those service accounts to access the cluster resources.
```

      key: "ca.crt"
```

### user

#### serviceAccount

> The configured annotations and labels will be applied to each service account created by Okteto.

Reviewer: ```suggestion
The configured annotations and labels will be applied to every service account that Okteto creates for an user.
```

> These annotations and labels are additional to the ones already applied by Okteto.

Reviewer: ```suggestion
These annotations and labels are added to the ones applied by Okteto.
```

Author: I've reworded this part, let me know what you think

> - `annotations`: Annotations applied to each service account generated by Okteto for user accounts.

Reviewer: @pchico83 should we also use the `extraAnnotations` pattern here like we do in role bindings to signify that this doesn't replace the base ones?

Reviewer: same for labels

Author: @rberrelleza you mean the key? this will imply re-opening the issue that was already approved and merged.

Reviewer 3: @rberrelleza in the helm world, everyone is using `labels` and `annotations` for extra labels/annotations. We are already using it for extra labels/annotations in namespaces, or our okteto components

Reviewer: @pchico83 I've seen both. Given that you can't replace the annotations and labels entirely, but add them, I think the extraLabels and extraAnnotations are a better name for the configuration. 

We already use that pattern on 'extraEnv' and 'extraRoleBindings' in our chart. Given that this is something new, I feel it is worth the extra engineering work to make it consistent before we expose it to customers.

Reviewer 3: We use `labels/annotations` in `namespace` which is a very use case than this one but for namespaces:
https://www.okteto.com/docs/self-hosted/administration/configuration/#namespace
And we do the same for extra labels/annotations on every one of our components:
- https://www.okteto.com/docs/self-hosted/administration/configuration/#api
- https://www.okteto.com/docs/self-hosted/administration/configuration/#autoscaler
- https://www.okteto.com/docs/self-hosted/administration/configuration/#buildkit
- ...
Using `labels/Annotations` is more aligned to our current chart

Reviewer: ok.

> - `annotations`: Annotations applied to each service account generated by Okteto for user accounts.

Reviewer: I think this description is redundant with the first paragraph. maybe we remove the first sentence ("The configured annotations and labels will be applied to each service account created by Okteto." and we just leave the explanation for annotations and labels?

Author: I've reworded this part, let me know what you think

- `labels`: Labels applied to each service account generated by Okteto for user accounts.

```yaml
user:
  serviceAccount:
    annotations:
      custom.annotation/one: one
      custom.annotation/two: two
    labels:
      custom.label/one: one
      custom.label/two: two
```

> A common use case for custom service account annotations is to get access via IAM Roles or Workload Identity to create Cloud resources as part of your Okteto Manifest `deploy` commands.

Reviewer: Can we add links to our docs on IAM Roles and Workload Identity? (even if they are on the community site)
```suggestion
A typical use case for this is integrating with IAM Roles or Workload Identity to create Cloud resources as part of your Okteto Manifest `deploy` commands.
```

Author: you mean this reply ? https://community.okteto.com/t/support-for-gke-workload-identities-for-okteto-self-hosted/39/4

Author: i've reworded this sentence and added the link under `get access via IAM Roles or Workload Identity`

### serviceaccount

> Annotation and label customizations for `serviceaccounts`.

Reviewer 3: ```suggestion
Annotation and label customizations for Service Accounts.
```

> The configured annotations and labels will be applied to each `serviceaccount` created by Okteto.

Reviewer 3: ```suggestion
The configured annotations and labels will be applied to each Service Account created by Okteto for user accounts.
```

These annotations and labels are additional to the ones already applied by Okteto.

> - `annotations`: Annotations applied to each `serviceaccount` generated by Okteto.

Reviewer 3: ```suggestion
- `annotations`: Annotations applied to each Service Account created by Okteto for user accounts.
```

> - `labels`: Labels applied to each `serviceaccount` generated by Okteto.

Reviewer 3: ```suggestion
- `labels`: Labels applied to each Service Account generated by Okteto for user accounts.
```

    custom.label/two: two
```

> Use these custom annotations to add IAM roles to service accounts and enable those service accounts to access the cluster resources.

Reviewer 3: ```suggestion
A common use case for custom Service Account annotations is to get access via IAM Roles or Workload Identity to create Cloud resources as part of your Okteto Manifest `deploy` commands.
```
